File: services/backtests/clients/python/cryptellation_backtests/order.py
from datetime import datetime


class Order(object):
    def __init__(
        self,
        time: datetime,
        type: str,
        exchange: str,
        pair: str,
        side: str,
        quantity: float,
        price: float,
    ):
        self.time = time
        self.type = type
        self.exchange = exchange
        self.pair = pair
        self.side = side
        self.quantity = quantity
        self.price = price

File: svc/backtests/pkg/order/order.go
package order

import (
	"errors"
	"time"
)

var (
	ErrInvalidOrderQty = errors.New("invalid-order-quantity")
)

type Order struct {
	ID            uint64
	ExecutionTime *time.Time
	Type          Type
	Exchange      string
	Pair          string
	Side          Side
	Quantity      float64
	Price         float64
}

func (o Order) Validate() error {
	if err := o.Type.Validate(); err != nil {
		return err
	}

	if err := o.Side.Validate(); err != nil {
		return err
	}

	if o.Quantity <= 0 {
		return ErrInvalidOrderQty
	}

	return nil
}
